public/apps/configuration/panels/role-edit/test/role-edit-filtering.test.tsx

```diff
@@ -0,0 +1,975 @@
+ *   permissions and limitations under the License.
+ */
+
+import React from 'react';
+import { ClusterPermissionPanel } from '../cluster-permission-panel';
+import { RoleEdit } from '../role-edit';
+import { ActionGroupItem } from '../../../types';
+import { fetchActionGroups } from '../../../utils/action-groups-utils';
+
+import { render, waitFor } from '@testing-library/react';
+
+import { act } from 'react-dom/test-utils';
+import { IndexPermissionPanel } from '../index-permission-panel';
+
+jest.mock('../../../utils/role-detail-utils', () => ({
+  getRoleDetail: jest.fn().mockReturnValue({
+    cluster_permissions: [],
+    index_permissions: [],
+    tenant_permissions: [],
+    reserved: false,
+  }),
+  updateRole: jest.fn(),
+}));
+jest.mock('../../../utils/action-groups-utils');
+
+jest.mock('../cluster-permission-panel', () => ({
+  ClusterPermissionPanel: jest.fn(() => null) as jest.Mock,
+}));
+
+jest.mock('../index-permission-panel', () => ({
+  IndexPermissionPanel: jest.fn(() => null) as jest.Mock,
+}));
+
+describe('Role edit filtering', () => {
+  const sampleSourceRole = 'role';
+  const mockCoreStart = {
+    http: 1,
+  };
+
+  it('basic cluster permission panel rendering', async () => {
+    const action = 'create';
+    const buildBreadcrumbs = jest.fn();
+
+    (fetchActionGroups as jest.Mock).mockResolvedValue({
+      data_access: {
+        reserved: true,
+        hidden: false,
+        allowed_actions: ['indices:data/*', 'crud'],
+        type: 'index',
+        description: 'Allow all read/write operations on data',
+        static: true,
+      },
+      cluster_manage_pipelines: {
+        reserved: true,
+        hidden: false,
+        allowed_actions: ['cluster:admin/ingest/pipeline/*'],
+        type: 'cluster',
+        description: 'Manage pipelines',
+        static: true,
+      },
+    });
+
+    render(
+      <RoleEdit
+        action={action}
+        sourceRoleName={sampleSourceRole}
+        buildBreadcrumbs={buildBreadcrumbs}
+        coreStart={mockCoreStart as any}
+        navigation={{} as any}
+        params={{} as any}
+        config={{} as any}
+      />
+    );
+
+    await act(async () => {
+      await waitFor(() => {
+        expect(ClusterPermissionPanel).toHaveBeenCalled();
+      });
+    });
+
+    const lastCallArgs =
+      ClusterPermissionPanel.mock.calls[ClusterPermissionPanel.mock.calls.length - 1];
+    const [props] = lastCallArgs;
+
+    // Cluster Permission Panel props is filtered to action groups with type cluster, and only the cluster permission constants
+    expect(props.optionUniverse).toEqual([
+      {
+        label: 'Permission groups',
+        options: [
+          {
+            label: 'cluster_manage_pipelines',
+          },
+        ],
+      },
+      {
+        label: 'Cluster permissions',
+        options: [
+          {
+            label: 'cluster:admin/component_template/delete',
```

Comment:
Is it possible to populate this using the value from [CONSTANTS](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/public/apps/configuration/constants.tsx#L33)?